libs/utils/src/http/endpoint.rs

```diff
@@ -63,8 +63,26 @@ async fn prometheus_metrics_handler(_req: Request<Body>) -> Result<Response<Body
     Ok(response)
 }
 
+pub fn add_request_id_middleware<B: hyper::body::HttpBody + Send + Sync + 'static>(
+) -> Middleware<B, ApiError> {
+    Middleware::pre(move |mut req| async move {
+        let headers = req.headers_mut();
+        let name = HeaderName::from_str("UUID").expect("created header name");
+        let request_id = uuid::Uuid::new_v4().to_string();
+        let value = HeaderValue::from_str(&request_id).unwrap();
+        headers.insert(name, value);
```

Comment:
Also I think we need to insert header only if it is missing. It will allow client to specify the id and it can be used to have end to end request id